samza-core/src/main/scala/org/apache/samza/container/TaskInstance.scala

```diff
@@ -218,6 +221,15 @@ class TaskInstance(
     trace("Checkpointing offsets for taskName: %s" format taskName)
 
     offsetManager.writeCheckpoint(taskName, checkpoint)
+
+    if (checkpoint != null) {
+      checkpoint.getOffsets.asScala
+        .filter { case (ssp, _) => committedMessagesDeletableStreams.contains(ssp.getStream) } // Only delete data of intermediate streams
```

Comment:
Minor: Faster to iterate over the streams with committedMessagesDeletable and look up in checkpointedOffsets.

Reply:
I am not sure how to achieve this given that `checkpoint.getOffsets` is keyed by SystemStreamPartition. Could you provide example code? Thanks.

Reply:
Never mind, types weren't clear from this snippet.

That reminds me, streamId might not be the same as the system stream name. I think what you need here is the physical name corresponding to the stream id. Can you check if this does the right thing if streamId != physical name?

Reply:
@prateekm @jmakes Thanks. I have updated the patch as suggested.